web-app-router/components/Input/index.tsx

```diff
@@ -1,7 +1,99 @@
-export const Input = () => {
+"use client";
+import clsx from "clsx";
+import { FloatingLabel } from "flowbite-react";
+import { ComponentProps, memo } from "react";
+import { Icon } from "../Icon";
+
+type InputProps = ComponentProps<"input"> & {
+  type?: ComponentProps<"input">["type"];
+  disabled?: boolean;
+  helperText?: string;
+  label: string;
+  className?: string;
+  placeholder?: string;
+  required?: boolean;
+  canCopy?: boolean;
+};
+
+export const Input = memo(function Input(props: InputProps) {
+  const {
+    helperText,
+    label,
+    placeholder,
+    className,
+    required,
+    disabled,
+    canCopy,
+    onChange,
+    ...restProps
+  } = props;
+
+  const customTheme = {
+    input: {
+      default: {
+        outlined: {
+          md: clsx(
+            "border-1 peer block w-full appearance-none rounded-lg border-gray-200 bg-transparent px-4 py-4 text-sm placeholder:text-gray-400",
+            "focus:border-blue-500 focus:text-gray-700 focus:outline-none focus:ring-0",
+            "disabled:bg-gray-50 disabled:text-gray-300",
+            { "pr-10": canCopy }
+            // Adjust right padding when canCopy is true
+          ),
+        },
+      },
+    },
+    label: {
+      default: {
+        outlined: {
+          md: clsx(
+            "absolute left-4 top-2 z-10 origin-[0] -translate-y-4 scale-75 transition-transform bg-white px-1 text-sm duration-300 scale-75 -translate-y-4",
+            "text-gray-400 peer-focus:text-blue-500",
+            disabled && [
+              "before:content-[''] before:absolute before:inset-0 before:bg-white before:z-[-1] before:rounded-lg before:w-full before:h-full",
+              "after:content-[''] after:absolute after:inset-x-0 after:bottom-0 after:bg-gray-50 after:z-[-1] after:rounded-b-lg after:w-full after:h-1/2",
+              "bg-transparent",
+            ],
+            {
+              "after:content-['*'] after:text-error-500 after:pl-1": required,
+            }
+          ),
+        },
+      },
+    },
+    helperText: {
+      default: "mt-2 text-xs text-gray-500 ",
+    },
+  };
+
   return (
-    <div>
-      <h1>Input</h1>
+    <div className={clsx(className, "relative")}>
+      <FloatingLabel
+        label={label}
+        theme={customTheme}
+        variant="outlined"
+        onChange={onChange}
+        sizing="md"
+        helperText={helperText}
+        placeholder={placeholder}
+        disabled={disabled}
+        required={true}
+      ></FloatingLabel>
+      {canCopy && (
```

Comment:
I'd prefer another approach here if this is possible. To have `addon: ReactNode` and `addonPosition: "left" | "right"` props to make it more flexible, because it's a common component.  For example if we will need to make something like eye button to hide/show password. 

I think we can implement it like this: 

```
<div className="grid grid-cols-[auto_1fr_auto]">
<div>{Boolean(addon) && addonPosition === "left" && addon}<div>
<input/>
<div>{Boolean(addon) && addonPosition === "right" && addon}<div>
</div>
```

If we are passing addon, it will take space of 1st or third grid cell, otherwise it will be 0px and only input will be visible.

And then we can use it like: 

```
<Input addon={<Component onClick={() => {something}}/> addonPosition="left"}
```

Reply:
I added the component in but requires some absolute style to do so. I think it could be better for the user to specify but lmk

Reply:
Hm, I can't say I like it. To adjust input side we need to use paddings the theme object as you did. What if addon will be a wide button that have width move than these paddings? Or what if passing an addon with more custom styles. Try to make button on a 3rd input as `flex` it breaks its position. 

My idea with the grid was that if we passing an addon it should take space by its sizes and it will automatically adjust input size. But as I can see there are no too many options to customize `FloatingLabel` in that way.

Reply:
Let's update input the way we discussed in Slack and then review again